### PR TITLE
Add Conditions CRD to deprecated features list

### PR DESCRIPTION
**Pull request by jerop**
# Changes

Update the deprecated features list to include `Conditions` CRD. Users should use `WhenExpressions` instead. 

PR: https://github.com/tektoncd/pipeline/pull/3135
TEP: https://github.com/tektoncd/community/blob/master/teps/0007-conditions-beta.md

The [API compatibility policy](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#alpha-beta-and-ga) states that Alpha features may be dropped at any time, though users will be given at least one release worth of warning.

So we're giving two releases warning on deprecation.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
    NONE
```

### CODE REVIEW

**Comment by bobcatfish** [comment]
Usually im all for separate PRs, but what do you think about making this change as part of https://github.com/tektoncd/pipeline/pull/3135? im thinking we wouldnt consider the feature deprecated until #3135 goes in anyway

**Comment by jerop** [comment]
@bobcatfish opened it separately because at the time #3135 would be merged, I wouldn't have the release announcement to reference on the deprecation table, so i'd still have to edit this table after #3135 is merged -- opened this as a draft now so that i don't forget to do it later

**Comment by bobcatfish** [comment]
ah kk, thanks for explaining @jerop !

**Review by vdemeester — APPROVED** [review]
Needs to be updated now that the dependent PR has been merged :upside_down_face:

File: docs/deprecations.md [hunk 1]
```diff
@@ -25,3 +25,4 @@ being deprecated.
...
 | [The `TaskRun.Status.ResourceResults.ResourceRef` field is deprecated and will be removed.](https://github.com/tektoncd/pipeline/issues/2694) | [v0.14.0](https://github.com/tektoncd/pipeline/releases/tag/v0.14.0)         | Beta                                                                                                     | April 30 2021                       |
 | [The `PipelineRun.Spec.ServiceAccountNames` field is deprecated and will be removed.](https://github.com/tektoncd/pipeline/issues/2614)       | [v0.15.0](https://github.com/tektoncd/pipeline/releases/tag/v0.15.0)         | Beta                                                                                                     | May 15 2021                         |
+| [`Conditions` CRD is deprecated and will be removed. Use `WhenExpressions` instead.](https://github.com/tektoncd/pipeline/pull/3135)       | [v0.16.0](https://github.com/tektoncd/pipeline/releases/tag/v0.16.0)         | Alpha                                                                                                     | Nov 02 2020                         |
```
Comment (jerop) [inline docs/deprecations.md:28]: The [API compatibility policy](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#alpha-beta-and-ga) states that Alpha features may be dropped at any time, though users will be given at least one release worth of warning. Is giving two releases worth of warning - September and October - okay? 
@vdemeester @bobcatfish

File: docs/deprecations.md [hunk 2]
```diff
@@ -25,3 +25,4 @@ being deprecated.
...
 | [The `TaskRun.Status.ResourceResults.ResourceRef` field is deprecated and will be removed.](https://github.com/tektoncd/pipeline/issues/2694) | [v0.14.0](https://github.com/tektoncd/pipeline/releases/tag/v0.14.0)         | Beta                                                                                                     | April 30 2021                       |
 | [The `PipelineRun.Spec.ServiceAccountNames` field is deprecated and will be removed.](https://github.com/tektoncd/pipeline/issues/2614)       | [v0.15.0](https://github.com/tektoncd/pipeline/releases/tag/v0.15.0)         | Beta                                                                                                     | May 15 2021                         |
+| [`Conditions` CRD is deprecated and will be removed. Use `WhenExpressions` instead.](https://github.com/tektoncd/community/blob/master/teps/0007-conditions-beta.md)       | [v0.16.0](https://github.com/tektoncd/pipeline/releases/tag/v0.16.0)         | Alpha                                                                                                     | Nov 02 2020                         |
```
Comment (jerop) [inline docs/deprecations.md:28]: The [API compatibility policy](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#alpha-beta-and-ga) states that Alpha features may be dropped at any time, though users will be given at least one release worth of warning. 

Is giving two releases worth of warning - September and October - okay? @vdemeester @bobcatfish

Comment (vdemeester) [reply docs/deprecations.md:28]: I think so :thinking: It is used in a `v1beta1` field though (refering to a `v1alpha1` object), but those fields should be considered alpha, so lgtm :+1:

**Pull request merged**